sdk/core/src/request_options/content_range.rs

```diff
@@ -0,0 +1,111 @@
+use crate::errors::ParseError;
+use std::fmt;
+use std::str::FromStr;
+
+const PREFIX: &str = "bytes ";
+
+#[derive(Debug, Copy, Clone, PartialEq)]
+pub struct ContentRange {
+    start: u64,
+    end: u64,
+    total_length: u64,
+}
+
+impl ContentRange {
+    pub fn new(start: u64, end: u64, total_length: u64) -> ContentRange {
+        ContentRange {
+            start,
+            end,
+            total_length,
+        }
+    }
+
+    pub fn start(&self) -> u64 {
+        self.start
+    }
+
+    pub fn end(&self) -> u64 {
+        self.end
+    }
+
+    pub fn total_length(&self) -> u64 {
+        self.total_length
+    }
+
+    pub fn is_empty(&self) -> bool {
+        self.end == self.start
+    }
+}
+
+impl FromStr for ContentRange {
+    type Err = ParseError;
+    fn from_str(s: &str) -> Result<ContentRange, ParseError> {
+        let remaining = match s.strip_prefix(PREFIX) {
+            Some(s) => s,
+            None => return Err(ParseError::TokenNotFound(PREFIX.to_owned())),
+        };
+
+        let v = remaining.split('-').collect::<Vec<_>>();
+        if v.len() != 2 {
+            return Err(ParseError::SplitNotFound('-'));
+        }
+
+        let start = v[0].parse()?;
+
+        let v = v[1].split('/').collect::<Vec<_>>();
```

Comment:
2 nits: 
* The name `v` is very non-descriptive and used quite a bit over nearly 15 lines of code. It is also shadowed. I would prefer a more descriptive name. 
* I would just use the iterator itself instead of allocating a vector. It's roughly the same amount of code but avoids an allocation.

Reply:
Good idea, especially not allocating is nice! 👍